src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -862,10 +862,91 @@ private function step_in_body() {
 				// Execution should not reach here; if it does then something went wrong.
 				return false;
 
-			default:
+		}
+
+		/*
+		 * These tags require special handling in the 'in body' insertion mode
+		 * that has not been implemented yet.
+		 *
+		 * As they're implemented, they should be removed from this list. An accompanying
+		 * test should help ensure this list is maintained.
+		 *
+		 * @see Tests_HtmlApi_WpHtmlProcessor::test_step_in_body_fails_on_unsupported_tags
+		 *
+		 * We throw the WP_HTML_Unsupported_Exception so we're free to implememnt
+		 * the catch-all handling for any other start and end tag.
+		 *
+		 * @see https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-inbody
+		 */
+		switch ( $tag_name ) {
+			case 'APPLET':
+			case 'AREA':
+			case 'BASE':
+			case 'BASEFONT':
+			case 'BGSOUND':
+			case 'BODY':
+			case 'BR':
+			case 'CAPTION':
+			case 'COL':
+			case 'COLGROUP':
+			case 'DD':
+			case 'DT':
+			case 'EMBED':
+			case 'FORM':
+			case 'FRAME':
+			case 'FRAMESET':
+			case 'HEAD':
+			case 'HR':
+			case 'HTML':
+			case 'IFRAME':
+			case 'INPUT':
+			case 'KEYGEN':
+			case 'LI':
+			case 'LINK':
+			case 'LISTING':
+			case 'MARQUEE':
+			case 'MATH':
+			case 'META':
+			case 'NOBR':
+			case 'NOEMBED':
+			case 'NOFRAMES':
+			case 'NOSCRIPT':
+			case 'OBJECT':
+			case 'OL':
+			case 'OPTGROUP':
+			case 'OPTION':
+			case 'PARAM':
+			case 'PLAINTEXT':
+			case 'PRE':
+			case 'RB':
+			case 'RP':
+			case 'RT':
+			case 'RTC':
+			case 'SCRIPT':
+			case 'SELECT':
+			case 'SOURCE':
+			case 'STYLE':
+			case 'SVG':
+			case 'TABLE':
+			case 'TBODY':
+			case 'TD':
+			case 'TEMPLATE':
+			case 'TEXTAREA':
+			case 'TFOOT':
+			case 'TH':
+			case 'THEAD':
+			case 'TITLE':
+			case 'TR':
+			case 'TRACK':
+			case 'UL':
+			case 'WBR':
+			case 'XMP':
 				$this->last_error = self::ERROR_UNSUPPORTED;
 				throw new WP_HTML_Unsupported_Exception( "Cannot process {$tag_name} element." );
```

Comment:
Would we like a different error message here?

Reply:
what would you think would be better? at this time it seems sufficient. the actual exception message never surfaces. vaguely I'd like to do that in some way; not necessary for this patch though.

Reply:
I was wondering if it would be helpful to have this be unique, so we can understand this error as "explicitly not handled" and the error below as "fallthrough case not handled".

Reply:
do what you feel is best. I don't consider it that important either way since this is only for debugging (it's not exposed to the end-user). either way, it's not supported, and so far that's the only thing that truly matters.